src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Entry<K, V>[] buckets;
+    private double threshold;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Entry[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
```

Comment:
```suggestion
        int index = getIndex(key);
        if (buckets[index] == null) {
            buckets[index] = new Node<>(key, value);
        } else {
            Node<K, V> current = buckets[index];
            if (Objects.equals(current.key, key)) {
                current.value = value;
                return;
            }
            while (current.nextNode != null) {
                if (Objects.equals(current.key, key)) {
                    current.value = value;
                    return;
                }
                current = current.nextNode;
            }
            current.nextNode = new Node<>(key, value);
        }
        size++;
```

Reply:
I've texted you another one solution of method put. Not required you to rewrite your code but there are some more clearly checks and logic

Reply:
Alright, I will try this solution as well in my free time and debug it to have a closer look. Thanks!